### PR TITLE
Readds Birdboatstation (ORANGES SAYS DO NOT TEST MERGE)

### PR DESCRIPTION
**Pull request by Bawhoppen**
:cl: Bawhoppen & Tokiko
add: Birdboatstation is back.
/:cl:

After numerous hours of updating paths and manual fixes, Birdboatstation (the best map ever made) is back. 
Birdboat is a lowpop map, originally made by Tokiko, which boasts a unique paranoia-laden design (TM) and several job modifications. 
The new version includes various updates and modernizations, such as techfabs and a working arrivals shuttle, among other things. It's likely missing some different newer features, but those can be added in as they're discovered.

FastDMM Map Image: https://www.dropbox.com/s/3sd4ly4zeskpcgm/Birdboat2Image3.png?dl=0
Old 2015 picture for reference: https://tgstation13.org/wiki//images/9/92/Birdstation1.png

### CODE REVIEW

**Comment by 81Denton** [comment]
My god, the horror

I'll review this in a bit

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Cyberboss) [inline _maps/map_files/BirdStation/job_changes.dm]: use the MAP_DISABLE_JOB define

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Cyberboss) [inline _maps/map_files/BirdStation/job_changes.dm]: muda

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Cyberboss) [inline _maps/map_files/BirdStation/job_changes.dm]: MAP_JOB_CHECK

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Cyberboss) [inline _maps/map_files/BirdStation/job_changes.dm]: MAP_JOB_CHECK

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Cyberboss) [inline _maps/map_files/BirdStation/job_changes.dm]: MAP_JOB_CHECK

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Cyberboss) [inline _maps/map_files/BirdStation/job_changes.dm]: MAP_JOB_CHECK

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by fludd12** [comment]
Yessss! My favorite map!

**Comment by FantasticFwoosh** [comment]
This was unironically my favourite map out of the mainstay box & meta too.

**Comment by zamolxius** [comment]
@Carbonhell thoughts? add it to hippie please

**Comment by 81Denton** [comment]
Few minor issues:

Engineering:
* The SME gas filters need to be rotated - right now, selecting a gas will cause it to be vented from the loop. This is the opposite from all other maps. Players will most likely just select nitrogen on both filters and have the engine delaminate in five minutes.
* Speaking of nitrogen, Engineering could use some canisters to start or just fix the crystal. 
* There is no supermatter monitoring console
* There aren't any radsuits in engineering
* The SME chamber needs an airlock or two, r-plasmaglass windows and less vents (2 scrubbers+2 vents would probably be best).
* I'd replace the supermatter crystal with a shard, otherwise half the map will be toast when it blows up.
* Atmos could use a gas mixer so that you can pump a proper airmix into distro.

Cargo:
* Cargo needs a bounty console
* Vault needs an ore silo and cash terminal
* The cargo conveyor belts are going the wrong direction and don't work with the shuttle's. So far it's like picture 1, but would probably be better as picture 2:
https://i.imgur.com/Jud94W1.png
https://i.imgur.com/grS4fP4.png
* They could probably use airtight flaps as well.
* Conveyor belts at the bottom of cargo are facing the opposite direction of the arrows next to them.

This map is gonna be hilarious. I'd probably also add `/obj/effect/spawner/lootdrop/two_percent_xeno_egg_spawner` to the xenobio cage at the very east of the station.

**Comment by Cyberboss** [comment]
This can be left open and test merged/etc for Halloween. I'm trying to finish the out-of-repo map system, though. so I'm not sold on merging it fully

**Comment by fludd12** [comment]
@81Denton 
>I'd replace the supermatter crystal with a shard, otherwise half the map will be toast when it blows up.

Even more reason to not let that happen! :bird::boat::boom:

**Comment by ShizCalev** [comment]
image

**Comment by 81Denton** [comment]
By the way, you can use FastDMM to easily generate a screenshot of the whole map

**Comment by Cyprex** [comment]
Why isn't mapdiffbot running?

**Comment by Bawhoppen** [comment]
@81Denton 
-Don't have a doctorate in atmos, so I will have to spend some time learning on how the SME filter needs to be set up. Or I could just leave it as is and make the map be different, as well as preserve Tokiko's original design vision.
-Will add a canister.
-Will add a console.
-There are radsuit lockers in the secure storage (surprisingly).
-Will add the windows and such.
-Shard, good idea probably.
-Originally Birdboat didn't actually even have an atmos, but eventually Tokiko caved to pressure and added a miniaturized one. This is why there is some missing atmos features.
-Will add bounty console.
-Will add silo and terminal.
-I'll look at the conveyors. I think the custom cargo shuttle itself's conveyors may also be broken already too.
-Will look at the flaps also.
-Will fix other conveyors.

Thanks for the review.

@ShizCalev Didn't know that was an issue. Those must've lingered from when the old paths were replaced. Will fix it.

**Comment by Farquaar** [comment]
Some quick questions/concerns, since I'm not very familiar with SS13 development. Keep in mind, I'm only looking at the image, so many of these points may have already been addressed.

Kitchen: It would be nice to have a snack cart or ice cream cart (maybe start off in maint?) It might also be helpful to give the chef some paper, a label maker, and wrapping paper.  Some people might complain about the lack of a deep fryer, but honestly I've never been a fan of it anyway.

Botany: Probably could use a smartfridge on the exterior facing the kitchen. Botanists are unlikely to deliver crops to the chef's smartfridge if it's that far away. I don't want this to become another Omega kitchen-botany situation. Also, the image doesn't show a gene manipulator, so that should be added if it isn't already.

Medical: Given the lack of a CMO office, there should probably be a medical hardsuit somewhere in the department.


On a positive note, this station continues to look supremely comfy. I especially love the theater, library, bar and garden areas.

**Comment by coiax** [comment]
Where did the murder grinder in disposals go to?

**Comment by coiax** [comment]
Also, is it possible to have Birdboat be built into red asteroids, rather than the grimdark lavaland rock?

**Comment by epochayur** [comment]
birdboat is so gd fun, but now the horrifying map design is thematic as well!

**Comment by Bawhoppen** [comment]
@coiax It was removed when I updated the paths. I intend to readd it when I do my suggestions passthrough.

File: _maps/map_files/BirdStation/job_changes.dm
Comment (81Denton) [inline _maps/map_files/BirdStation/job_changes.dm]: You'll need to remove `ACCESS_CARGO_BOT` and the other access defines that were axed in #40892

**Comment by Bawhoppen** [comment]
@coiax Actually, I see what you mean now. I just updated a random version that I had saved from October 2015, which may not have had it yet.

**Comment by Bawhoppen** [comment]
Alright, I've implemented all the suggestions I feel fit with the original design philosophy of the map.

I didn't make the rocks red, even though I'd like to because it would be quite a bit of work.
Tcomms is also still a mainframe, rather than proper machines, something which I intend to change later. In addition to those things, I've also set a maxplayers of 50, which is probably beyond the ideal pop, but should be reasonable.

Should be ready for testmerging (or real merging if Cyber changes their mind) now!

**Comment by Bawhoppen** [comment]
OP updated with image of the map from FastDMM!

**Comment by Borisvanmemes** [comment]
Testmerged on sybil right now and i can't seem to use the change-map verb to force it.

**Comment by Bawhoppen** [comment]
I ended up waiting all those hours, only for it not to work during the testmerge. I have to guess it's because of the config element.

**Comment by Arianya** [comment]
Yeah it was a miscommunication between us and MSO on how to testmerge it such that it could be NextMap'd but not be in rotation. I'll update the config later today so that it can be NextMap'd, apologies for the inconvenience.

**Comment by Bawhoppen** [comment]
@Arianya Oh, great. Thanks for the help.

**Comment by Arianya** [comment]
The config change should be done so let me know if it still doesn't work for some reason.

File: _maps/map_files/BirdStation/job_changes.dm
Comment (SpaceManiac) [inline _maps/map_files/BirdStation/job_changes.dm]: This is causing HOP to have no access on all maps except BirdStation, look at the OmegaStation removal for what it should be

**Comment by zeroisthebiggay** [comment]
there are no holopads

**Comment by Suicidalpickles** [comment]
Station could probably use some intercomms, AI holopads, and more O2 lockers scattered about as well.

**Comment by Bawhoppen** [comment]
@zeroisthebiggay The lack of holopads is from the original version. I don't want to add a ton, but I suppose I could add a few.
@Suicidalpickles Original birdboat also didn't feature any intercomms, but in replacement you had custom emergency boxes that contained a radio. I scrapped this so I could get the PR up faster. I'll see about readding them. The lack of O2 lockers is sort of a feature, since this is supposed to be bare minimum safety standards. I'll review if there are any areas that especially need them, however.

**Comment by Bawhoppen** [comment]
Fixed some of the issues that became apparent during the overly long playtesting on Sybil.

Something odd to note, likely due to the testmerge, is that the disabled jobs are not working. This causes people who select these supposed-to-be disabled jobs to spawn in random spawnpoints around the map. Not sure what to do to fix this.

**Comment by Arianya** [comment]
I'm no coder but testmerging it shouldn't make a difference to that. Testmerged code is functionally identical to fullmerged as far as BYOND is concerned (AFAIK)

**Comment by Bawhoppen** [comment]
Ah, nevermind. Sorry, it was my bad. Must not have tested it after editing it.

**Comment by Cyberboss** [comment]
Let me just state that I have no intention of merging this because I'm trying to finalize the prototype for the new out-of-repo maploading system that will eventually be used for all the current station maps save Meta and Runtime.

This isn't a dismissal of the PR by any means though, feel free to keep getting it test merged.

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Cyberboss) [reply _maps/map_files/BirdStation/job_changes.dm]: This needs to be corrected however
```dm
. = ..()
MAP_JOB_CHECK
return get_all_accesses()
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Cyberboss) [inline _maps/map_files/BirdStation/job_changes.dm]: This needs to be the same as `map_name` in the JSON iirc

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Bawhoppen) [reply _maps/map_files/BirdStation/job_changes.dm]: Oh, that's probably why it wasn't working. Makes sense.

File: _maps/map_files/BirdStation/job_changes.dm
Comment (Bawhoppen) [reply _maps/map_files/BirdStation/job_changes.dm]: Are you sure it's not supposed to be MAP_JOB_CHECK_BASE like how Omega did it though?

**Comment by optimumtact** [comment]
@Cyberboss any day now

**Comment by Bawhoppen** [comment]
I gotta remove the circuit area I added, don't let me forget please.

**Comment by miniusAreas** [comment]
@Bawhoppen I duno man. The creater of birdboat requested that the map stay dead according to the wiki, but non the less its a welcome change.

**Comment by optimumtact** [comment]
@Cyberboss I'm only going to hold this a few more days for you

**Comment by Bawhoppen** [comment]
@miniusAreas Tokiko wanted it removed because it received a lot of criticism from the community, mainly because the wrong type of crowd was playing it. I've limited the pop to 50 players, and fixed some of the more egregious issues I think, so it should be better received this time by a broad population hopefully.

**Comment by Bawhoppen** [comment]
Should be ready for merge at oranges' discretion. I did do Cyber's review, even though it still shows as current.

**Comment by Cyberboss** [comment]
Eh, go ahead then, can always be fixed later

**Comment by Bawhoppen** [comment]
So what is the status on this being merged? It's been ready for for awhile, though not testmerged much even though I asked a few times. I completed Cyber's review, though it still shows as unresolved.
(Am I supposed to put [READY] in the title?)

**Comment by Jordie0608** [comment]
The map doesn't pass our tests, you need to remove all the tag vars.

**Comment by Bawhoppen** [comment]
Why did all these issues not show up the first times Travis ran? Weird.

**Comment by Cyberboss** [comment]
It's fail fast

**Comment by Bawhoppen** [comment]
Hey look, it passes Travis now. 
Should I have this testmerged more first before a real merge?

**Comment by Arianya** [comment]
Can I get some guesstimations on min (presumably any, since its a lowpop map) and max (this is important so we don't have 60 player birdboat rounds) population?

**Comment by Bawhoppen** [comment]
@Arianya I'd say the ideal pop range is in the 30s. It is a pretty small map, so I set the maxpop to be 50, though the original Tokiko version had it at 45. Should have no minimum pop.

**Comment by optimumtact** [comment]
```
_maps/map_files/BirdStation/BirdStation2.dmm:15955:error: undefined type: /obj/item/clothing/glasses/sunglasses/blindfold
_maps/map_files/BirdStation/BirdStation2.dmm:15957:error: unknown type
_maps/map_files/BirdStation/BirdStation2.dmm:16720:error: undefined type: /obj/item/clothing/glasses/sunglasses/blindfold
_maps/map_files/BirdStation/BirdStation2.dmm:16721:error: undefined type: /obj/item/clothing/glasses/sunglasses/blindfold
_maps/map_files/BirdStation/BirdStation2.dmm:16728:error: unknown type
_maps/map_files/BirdStation/BirdStation2.dmm:16728:error: unknown type
```

**Comment by Bawhoppen** [comment]
Didn't realise Travis wouldn't re-test until I pushed a commit, so I didn't see those there earlier.

**Comment by Bawhoppen** [comment]
I'm gonna seek to have this testmerged again so that way we can finally have it merged.

**Comment by fludd12** [comment]
I'm baw-hopin' for this to get merged

**Comment by subject217** [comment]
Greetings. I have travelled to this PR to inform you that you must remove the drone dispenser from your map. Thank you.

**Comment by Bawhoppen** [comment]
@subject217 I thought I did though...

**Comment by subject217** [comment]
image

**Comment by Bawhoppen** [comment]
Oh. I didn't even realise that one existed, so I only removed the other one.

**Comment by Names-Are-Hard** [comment]
I've noticed that people seem to spawn in areas they don't have the access to leave. For example: doctors spawning in engineering.

**Comment by 81Denton** [comment]
Yeah, there are also doctors spawning in the cap's office

**Comment by 81Denton** [comment]
Few things I noticed:

The max player limit should probably be lowered to something like ~30. The station's departments are just too small to sustain a large population. 

There are no vents/scrubbers and no distro/waste loop.

Arrivals has the "Change'e shrine" which just ended up being two wooden walls. Is it missing textures or was it always like that?

The AI core layout is a bit weird; the core is exposed and there is an airlock that serves no real purpose:
qfweffefe

**Comment by 81Denton** [comment]
Oh the map also doesn't have firedoors

**Comment by Cyprex** [comment]
Could the map be removed from testmerge until the above issues are fixed? The more egregious ones like captains office at least.

**Comment by anconfuzedrock** [comment]
Birdboat's grinder is supposed to start emagged cmon guys you forgot the important part

**Comment by Bawhoppen** [comment]
@Names-Are-Hard @81Denton 
Is that only happening for doctors? They have multiple spawning landmarks, so it shouldn't be an issue. I'll try adding more though.

@81Denton 
-I totally agree that 30 or so would be the ideal pop, but I wanted to leave some flexibility in it. It didn't seem like 50 would be awful, but if it actually is, I'll lower it back to 45 (original pop limit) or even to 40.

-Tokiko designed the station to have only the bare necessities, so it doesn't have a mainline atmos system (in the most original versions didn't have ANY atmos system even). I would like to follow through with the same design philosophy, but I don't think it would be giving too much if I made a few basic vents in public areas around the station. 

-I was never actually sure what the "Change'e shrine" was really about, but that's definitely how it always looked at least. 

-I think that door is a relic from a previous design. I'll fix the core design.

-The lack of firedoors is part of the whole "bare-minimum metal-deathtrap" design philosophy. Right now, there are a few shutters in the hallways though that are for fires I believe. Adding in a few actual firelocks in some select locations may be acceptable though.

@Electronics 
-Will fix the duplicate spares.

-Will look into the remotes and fix.

-Will add cycle-links for dual doors.

-No idea why the pinpointer doesn't work. Will look into.

@tortellinitony 
In what way? I don't know very much about NTnet.

**Comment by carshalash** [comment]
Is this just gonna keep getting testmerged despite how awful it is?

**Comment by carshalash** [comment]
There is no atmos system, a single fire can fuck things forever.  
AI has no intercomms at all in its core.
Half of the fucking jobs are missing but we have enough space for all of these maint monstrosities 

There is nothing good about this station, It's always going to just be the meme options in polls or something admins force when they're mad at players, we have fucking events like meteors and immovable rods in the game which breach the station as a core part of the game now, you can't just go "WELL IT USED TO BE LIKE THIS BACK IN THE OLD DAYS" as an excuse to ignore the very blatant problem of there being no atmos system in game.

**Comment by zeroisthebiggay** [comment]
things i see that are problems

-drones (not a problem with me but admemes probably aren't happy)
-no vents (why)
-no scrubbers (w h y)
-no firelocks (**w h y**)
-no intercoms literally why
-the supermatter makes no sense
-the teg is backasswards
-security lockers have no secbelts in them
-like the old days =/= good

**Comment by funnyman2003** [comment]
C U R S E D

**Comment by Trazen4** [comment]
Please no.

**Comment by tortellinitony** [comment]
Replace the recycler with `/obj/machinery/recycler/deathtrap`

**Comment by Cyprex** [comment]
I'll check out atmos, portable scrubbers seem like they could be used in place of scrubbers, as can air canisters, if atmos can make them easily.

**Comment by optimumtact** [comment]
For the love of god stop test merging this disaster

**Comment by Bawhoppen** [comment]
The testmerge unfortunately doesn't get updated very frequently with fixes, which is part of the problem.

**Comment by FloranOtten** [comment]
Either the HoP office floor is slanted, or these stairs form a way to make an infinite waterfall

image

**Comment by optimumtact** [comment]
I am concerned about the number of maps and people seem to dislike this one.

**Comment by Bawhoppen** [comment]
Closing is probably for the best. I am sadly still unable to work on fixing the map's current issues atm. If anyone wants to pick up working on this map, you have my blessing.

**Pull request closed without merging**
